unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -25,26 +25,138 @@ export function buildCategoryFacetResults(numberOfResults = 11, numberOfRequeste
 
 export function CategoryFacetTest() {
   describe('CategoryFacet', () => {
+    let options: ICategoryFacetOptions;
     let test: IBasicComponentSetup<CategoryFacet>;
     let simulateQueryData: ISimulateQueryData;
 
     beforeEach(() => {
+      options = { field: '@field' };
       simulateQueryData = buildCategoryFacetResults();
       initializeComponent();
     });
 
     function initializeComponent() {
       test = Mock.advancedComponentSetup<CategoryFacet>(
         CategoryFacet,
-        new Mock.AdvancedComponentSetupOptions(null, { field: '@field' }, env => env.withLiveQueryStateModel())
+        new Mock.AdvancedComponentSetupOptions(null, options, env => env.withLiveQueryStateModel())
       );
-      test.cmp.activePath = simulateQueryData.query.categoryFacets[0].path;
+      test.cmp.changeActivePath(simulateQueryData.query.categoryFacets[0].path);
+
+      spyOn(test.cmp.logger, 'warn');
+      spyOn(test.cmp, 'scrollToTop').and.callThrough();
+      spyOn(test.cmp, 'reset').and.callThrough();
     }
 
     function allCategoriesButton() {
       return $$(test.cmp.element).find('.coveo-category-facet-all-categories');
     }
 
+    describe('Testing collapse/expand', () => {
+      function validateExpandCollapse(shouldBeCollapsed: boolean) {
+        expect($$(test.cmp.element).hasClass('coveo-dynamic-category-facet-collapsed')).toBe(shouldBeCollapsed);
+      }
+
+      function initializeComponentWithCollapse(enableCollapse: boolean, collapsedByDefault: boolean) {
+        options.enableCollapse = enableCollapse;
+        options.collapsedByDefault = collapsedByDefault;
+        initializeComponent();
+        Simulate.query(test.env, simulateQueryData);
+      }
+
+      it(`when enableCollapse & collapsedByDefault options are true
+      facet should be collapsed`, () => {
+        initializeComponentWithCollapse(true, true);
+        validateExpandCollapse(true);
```

Comment:
Not a fan of those kind of method where you pass in true/false flag to do two different things, because it makes stuff kinda hard to read to know what `true/false` means in the context of the function.

Would be more descriptive to have something like `validateWasExpanded()` and `validateWasCollapsed()` .

I have a similar comment for the `initializeComponentWithCollapse` function.